### ISSUE TITLE
Table: adaptive column collapse (when the column width is 100%) 

### ISSUE THREAD

**Issue by a1eksandrk** [gravity-ui/uikit#1288]
Hi! 
If you set the width to 100 as shown in the storybook example, the column collapses at a small size, which I think is not quite the expected behavior. 
However, if you remove the `max-width: 0` property, the behavior changes to "more expected", the column does not collapse, but behaves as default, minimum width by content.

What do you think about it? What is the purpose of using `max-width: 0`?

![image](https://github.com/gravity-ui/uikit/assets/60098929/7d49e327-a312-4d5d-bb99-c3d14023298e)

**Comment by Raubzeug** [comment]
HI @AlexKomz !

**Comment by Raubzeug** [comment]
Hi @AlexKomz !
The key purpose of adding `maxWidth: 0` is avoiding cell stretching if content overflows. If we remove `maxWidth: 0`, it will looks like:
![2024-01-26 17 27 46](https://github.com/gravity-ui/uikit/assets/42094364/85b0e105-b78d-491d-9e1e-904936ff152c)

**Comment by AlexKomz** [comment]
Hi, @Raubzeug! Thank you for your reply
You can watch it here: https://preview.gravity-ui.com/uikit/?path=/story/components-data-display-table--adaptive

Attached screenshots for more clarity

![image](https://github.com/gravity-ui/uikit/assets/60098929/accb9a56-59ce-4eb1-b09b-688b082ef378)
![image](https://github.com/gravity-ui/uikit/assets/60098929/cf083ff5-bf7c-4b0b-aa2e-59c69678855b)

**Comment by AlexKomz** [comment]
I have to do workaround `max-width: min-content !important;` in this case to keep the content from collapsing. Is a more elegant solution possible? I was thinking of a PR to add a `minWidth` property for a column

**Comment by Raubzeug** [comment]
@AlexKomz thanks, now I see the problem. Unfortunately it seems that your workaround would let content to stretch the cell as I've shown above :(
I thinks that the most appropriate solution will be to pass the className property to the column with `min-width` in pixels.

**Comment by AlexKomz** [comment]
@Raubzeug 
Yes, that's what I'm doing as I wrote in a previous post. But of course it is confusing that I need to increase specificity with important, because there are inline styles there 🥲

**Comment by Raubzeug** [comment]
@AlexKomz 
But you should not. You should pass `min-width` property, it has no collisions with existing props.

**Comment by AlexKomz** [comment]
@Raubzeug
Yes, my mistake, I read `max` instead of `min` in the post 🤕 You're right.

**Comment by larannma** [comment]
@Raubzeug Hello, I'm trying to customize the width of the Table component but setting custom styles as width: 100% or more specific width: 800px for example isn't working. It works if i add it from the dev console but I cannot access component inside Table itself. Do you have any suggestions of how i can do this? Thank you!

**Comment by Raubzeug** [comment]
@larannma hi! The `Table` allows to pass [property](https://github.com/gravity-ui/uikit/blob/main/src/components/Table/Table.tsx#L171C13-L171C23) `width='max'` that will stretch Table to 100% width of its parent container. Will it be enough for your needs?

**Comment by larannma** [comment]
@Raubzeug It is exactly what i needed. Thank you !

**Comment by larannma** [comment]
@Raubzeug It is yes for me
